### PR TITLE
Replace update versions

### PR DESCRIPTION
**Pull request by viniciusdc**
Replace `update_upstream_versions` and some changes with `test_update_versions` (now with the changes proposed by @CJ-Wright  #1073 )

### CODE REVIEW

**Comment by viniciusdc** [comment]
Hi @beckermr if you have time, could you test this ? and also, how you test it ?(maybe this can help me to figure out a way to test the profile code too), thanks in advance 😄

File: conda_forge_tick/update_upstream_versions.py [hunk 1]
```diff
@@ -488,45 +42,72 @@ def get_latest_version(
...
+    node_count = 0
     to_update = []
     for node, node_attrs in _all_nodes:
-        attrs = node_attrs["payload"]
-        if attrs.get("bad") or attrs.get("archived"):
-            attrs["new_version"] = False
-            continue
-        to_update.append((node, attrs))
+        with node_attrs["payload"] as attrs:
+            if (attrs.get("bad") and "Upstream" not in attrs["bad"]) or attrs.get(
```
Comment (CJ-Wright) [inline conda_forge_tick/update_upstream_versions.py:77]:
```suggestion
            if "Upstream" not in attrs.get("bad", []) or attrs.get(
```

Comment (viniciusdc) [reply conda_forge_tick/update_upstream_versions.py:77]: Thanks CJ,I did not know that the `.get()` method could be used this way.

**Comment by viniciusdc** [comment]
BTW, as we are not using the graph to update, I thought to change the entry of the `_sequencial` and `_pool` functions to replace the `gx` variable by the 
```python
_all_nodes = [t for t in gx.nodes.items()]
random.shuffle(_all_nodes)
# do a loop here
with node_attrs["payload"] as attrs:
            if "Upstream" not in attrs.get("bad", []) or attrs.get("archived"):
                continue
            to_update.append((node, attrs))
```
and the pass the `to_update` as entry for these funtions (this way we make the separation just one time)

**Comment by CJ-Wright** [comment]
I'd put that into the PR so it is a bit easier to evaluate.

**Comment by viniciusdc** [comment]
> I'd put that into the PR so it is a bit easier to evaluate.

I will do that after we merge this (when everything is right), cause that can wait until we start the optimization process of the code.

**Comment by viniciusdc** [comment]
BTW @CJ-Wright all the changes to the files we have are only update during the deploy phse right, or is there some funtion that do it's own update ?

**Comment by CJ-Wright** [comment]
the deploy phase commits the changes and pushes them to github

**Comment by beckermr** [comment]
We'll need to stop the bot, merge this, change the circle scripts and then restart the bot

**Comment by beckermr** [comment]
This code in cli.xsh needs to be adjusted: https://github.com/regro/cf-scripts/blob/master/conda_forge_tick/cli.xsh#L16

**Comment by viniciusdc** [comment]
> the deploy phase commits the changes and pushes them to github

Yes, thanks. My fear was that some independent part of the bot has its own deploy.

**Comment by beckermr** [comment]
This line too. https://github.com/regro/cf-scripts/pull/1075/files/75e7c97c9a19add17a142c56ae9de0486a5a5eec..5dd0497d403c25fa87c08f21284d37e82156f4ac#diff-11f79692d3a38972cab98dbfb00dce17R93

**Comment by viniciusdc** [comment]
> This line too. https://github.com/regro/cf-scripts/pull/1075/files/75e7c97c9a19add17a142c56ae9de0486a5a5eec..5dd0497d403c25fa87c08f21284d37e82156f4ac#diff-11f79692d3a38972cab98dbfb00dce17R93

you want to point to this ` 7: main_populate_versions,` right ?

**Comment by beckermr** [comment]
We can remove it and adjust circle which is simpler and has to be done anyways.

File: conda_forge_tick/cli.xsh [hunk 2]
```diff
@@ -13,7 +13,6 @@ from .auto_tick import main as main_auto_tick
 from .status_report import main as main_status_report
 from .audit import main as main_audit
 from .update_prs import main as main_update_prs
-from .new_update_versions import main as main_populate_versions
```
Comment (CJ-Wright) [inline conda_forge_tick/cli.xsh:16]: Why was this removed? Don't we still need it in the CLI?

Comment (viniciusdc) [reply conda_forge_tick/cli.xsh:16]: We will replace the current one with the new, so this process is redundant now

Comment (viniciusdc) [reply conda_forge_tick/cli.xsh:16]: I think the ideia is to remove `update_versions` for the `make_graph` job, and them update the `update version` job at Circle using the `update_versions` instead

**Comment by viniciusdc** [comment]
Thanks Matt

**Comment by beckermr** [comment]
I am stopping the bot now so we can merge this.

File: conda_forge_tick/cli.xsh (hunk 2 above)
Comment (beckermr) [reply conda_forge_tick/cli.xsh:16]: yup.

**Comment by viniciusdc** [comment]
> I am stopping the bot now so we can merge this.

Right, you will update the circle config ?

**Comment by beckermr** [comment]
Once we merge this, we can merge this: https://github.com/regro/circle_worker/pull/151

**Comment by beckermr** [comment]
Yup I updated circle!

**Comment by beckermr** [comment]
@CJ-Wright You ok if we merge this?

**Comment by viniciusdc** [comment]
@CJ-Wright @beckermr IDK what happened [here](https://app.circleci.com/pipelines/github/regro/circle_worker/16657/workflows/25c0f002-1166-4a98-9875-3793ce7a6753/jobs/34175) (the code seems to be right), so I will wait for the next `hourly` job and see what happens.
